icaruscode/Decode/MCDecoderICARUSTPCwROI_module.cc

```diff
@@ -498,110 +499,54 @@ void MCDecoderICARUSTPCwROI::processSingleLabel(art::Event&
     // Require a valid handle
     if (digitVecHandle.isValid() && digitVecHandle->size()>0 )
     {
-        // Sadly, the RawDigits come to us in an unsorted condition which is not optimal for
-        // what we want to do here. So we make a vector of pointers to the input raw digits and sort them
-        std::vector<const raw::RawDigit*> rawDigitVec;
-
-        // Ugliness to fill the pointer vector...
-        for(size_t idx = 0; idx < digitVecHandle->size(); idx++) rawDigitVec.emplace_back(&digitVecHandle->at(idx)); //art::Ptr<raw::RawDigit>(digitVecHandle, idx).get());
-
-        // Sort (use a lambda to sort by channel id)
-        std::sort(rawDigitVec.begin(),rawDigitVec.end(),[](const raw::RawDigit* left, const raw::RawDigit* right) {return left->Channel() < right->Channel();});
-
-        // Declare a temporary digit holder and resize it if downsizing the waveform
-        unsigned int               dataSize = art::Ptr<raw::RawDigit>(digitVecHandle,0)->Samples(); //size of raw data vectors
-        raw::RawDigit::ADCvector_t rawDataVec(dataSize);
-
-        using BoardToChannelArrayPairMap = std::map<unsigned int, ChannelArrayPair>;
-
-        BoardToChannelArrayPairMap  boardToChannelArrayPairMap;
-        std::map<unsigned int, int> boardWireCountMap;
-        const unsigned int          MAXCHANNELS(64);
-
-        // Commence looping over raw digits
-        for(const auto& rawDigit : rawDigitVec)
-        {
-            raw::ChannelID_t channel = rawDigit->Channel();
-
-            ChannelToBoardWirePlaneMap::const_iterator channelToBoardItr = fChannelToBoardWirePlaneMap.find(channel);
-
-            if (channelToBoardItr == fChannelToBoardWirePlaneMap.end())
+        const unsigned int dataSize = art::Ptr<raw::RawDigit>(digitVecHandle,0)->Samples(); //size of raw data vectors
```

Comment:
How is this different from
```suggestion
        const unsigned int dataSize = digitVecHandle->front().Samples(); //size of raw data vectors
```
?

Reply:
I guess they are the same. Testing your version and if it works I will commit it